src/app/chat/views/sticker_list.nim

```diff
@@ -0,0 +1,49 @@
+import NimQml, Tables
+import ../../../status/chat/stickers
+
+import ../../../status/libstatus/types
+
+type
+  StickerRoles {.pure.} = enum
+    Url = UserRole + 1
+    Hash = UserRole + 2
+
+QtObject:
+  type
+    StickerList* = ref object of QAbstractListModel
+      stickers*: seq[Sticker]
+
+  proc setup(self: StickerList) = self.QAbstractListModel.setup
+
+  proc delete(self: StickerList) = self.QAbstractListModel.delete
+
+  proc newStickerList*(): StickerList =
+    new(result, delete)
+    result.stickers = @[]
+    result.setup()
+
+  method rowCount(self: StickerList, index: QModelIndex = nil): int = self.stickers.len
+
+  method data(self: StickerList, index: QModelIndex, role: int): QVariant =
+    if not index.isValid:
+      return
+    if index.row < 0 or index.row >= self.stickers.len:
+      return
+
+    let sticker = self.stickers[index.row]
+    let stickerRole = role.StickerRoles
+    case stickerRole:
+      of StickerRoles.Url: result = newQVariant(decodeContentHash(sticker.hash))
+      of StickerRoles.Hash: result = newQVariant(sticker.hash)
+
+  method roleNames(self: StickerList): Table[int, string] =
+    {
+      StickerRoles.Url.int:"url",
+      StickerRoles.Hash.int:"hash"
+    }.toTable
+
+  proc addStickerToList*(self: StickerList, sticker: Sticker): int =
+    self.beginInsertRows(newQModelIndex(), 0, 0)
+    self.stickers.insert(sticker, 0)
+    self.endInsertRows()
+    result = 0
```

Comment:
the 0 return here isn't necessary

Reply:
This method was no longer needed with the modification to the constructor that accepts a `seq[Sticker]`.